src/math/p5.Vector.js

```diff
@@ -326,6 +326,141 @@ p5.Vector.prototype.add = function add(x, y, z) {
   return this;
 };
 
+/**
+ * Gives remainder of a vector when it is divided by another vector.
+ * See examples for more context.
+ *
+ * @method rem
+ * @param {Number} x the x component of divisor vector
+ * @param {Number} y the y component of divisor vector
+ * @param {Number} z the z component of divisor vector
+ * @chainable
+ * @example
+ * <div class='norender'>
+ * <code>
+ * let v = createVector(3, 4, 5);
+ * v.rem(2, 3, 4);
+ * // v's components are set to [1, 1, 1]
+ * </code>
+ * </div>
+ * <div class="norender">
+ * <code>
+ * // Static method
+ * let v1 = createVector(3, 4, 5);
+ * let v2 = createVector(2, 3, 4);
+ *
+ * let v3 = p5.Vector.rem(v1, v2);
+ * // v3 has components [1, 1, 1]
+ * print(v3);
+ * </code>
+ * </div>
+ */
+/**
+ * @method rem
+ * @param {p5.Vector | Number[]}  value  divisor vector
+ * @chainable
+ */
+p5.Vector.prototype.rem = function rem(x, y, z) {
+  const calculateRemainder3D = (xComponent, yComponent, zComponent) => {
```

Comment:
do these functions need to be defined within the function? it's preferable to define them outside to minimize execution time

Reply:
I think you could change these methods to accept an array of length 3 (or 2 for 2D function). As it is now, you end up needing to split up the arrays to call the functions below.

Reply:
> do these functions need to be defined within the function? it's preferable to define them outside to minimize execution time

- These functions are defined inside because they are borrowing **this** keyword from the parent scope. I will refactor them as helper functions outside the scope. Thanks for pointing that out as I read about inner functions and found that they hamper function execution time. I think the reason behind this might be that the execution state for the inner function is maintained due to closure. Am I thinking right?

Reply:
> I think you could change these methods to accept an array of length 3 (or 2 for 2D function). As it is now, you end up needing to split up the arrays to call the functions below.

Are you suggesting to write functions in below mentioned way?
```
/**
* @param Number[]
*  returns Number[]
*/
function calculateRemainder2D(array){
}
```